pkg/sysfs/system.go

```diff
@@ -306,6 +309,21 @@ type Cache struct {
 	cpus  idset.IDSet // CPUs sharing this cache
 }
 
+// cacheOverrides is a list of cache overrides for specific CPU sets.
+// Every CPU set gets cache id that starts from 0 on each override object.
+// Example: two 128 MB L3 Caches shared by CPUs 0-15,32-47 (socket 0)
+// and 16-31,48-63 (socket 1). The cache id is 0 for the first, 1 for the second.
+// [{"cpusets": ["0-15,32-47", "16-31,48-63"], "level": 3, "size": "128M"}]
+type cacheOverrides []cacheOverride
+type cacheOverride struct {
+	Cpusets []string // cpusets to which this override applies
```

Comment:
nit: what is the reason for allowing multiple cpusets strings here ? Isn't it always possible to present them as a single one ?

Reply:
Ah, ok. It is for being able to use different cache IDs for different CPU sets, which I guess is necessary to be able to emulate some hybrid core platforms...

Reply:
Yes, this gives a possibility to have quite realistic id numbering without forcing user to write a lot (by specifying ids), or a developer to write a lot (come up with a magic that combines CPU topology with caches). Something like:

```
[
  {"cpusets": ["0-7", "8-15"], "level": 3, "size": "128M"},
  {"cpusets": ["0-3", "4-7"], "level": 2, "size": "16M"},
  {"cpusets": ["8-11", "12-15"], "level": 2, "size": "16M"},
  {"cpusets": ["0-1", "2-3", "4-5", "6-7"], "level": 1, "size": "256k", "kind": "d"},
  {"cpusets": ["0-1", "2-3", "4-5", "6-7"], "level": 1, "size": "128k", "kind": "i"},
  {"cpusets": ["8-9", "10-11", "12-13", "14-15"], "level": 1, "size": "256k", "kind": "d"},
  {"cpusets": ["8-9", "10-11", "12-13", "14-15"], "level": 1, "size": "128k", "kind": "i"}
  ]
```